### ISSUE TITLE
bug: Add the possibility to insert an aria-modal on an alert

### ISSUE THREAD

**Issue by terrancle** [ionic-team/ionic#21204]
On an alert, you are currently using a role = "dialog", which is perfect.
However, it should be possible to have the choice of specifying to screen readers the aria-modal = "true" tag depending on the situation.
Can you add such a parameter? or declare an aria-modal = "true" on this type of alert, so that the screen reader cannot see the rest of the window, which disrupts navigation.

async presentAlert(header: string, subHeader: string, message: string) {
    const alert = await this.alertController.create({
      header: header,
      subHeader: subHeader,
      message: message,
      buttons: ['OK']
    });
    await alert.present();
  }

**Comment by liamdebeasi** [comment]
Thanks for the issue. I took a closer look at this, and it seems that elements with `role="dialog"` also inherit `aria-model="true"`. See: https://www.w3.org/TR/wai-aria-1.1/#dialog

Is there a particular issue you are running into here?

**Comment by terrancle** [comment]
Hello,

Thank you for your reply.

Currently, during an alert with the previous source code, VoiceOver sees the alert, but it also sees the window in the background.
It is for this reason that I asked you to add an aria-modal = "true", so that VoiceOver can only look at the alert window. He should only see the initial window when the "ok" or "close" button is activated.
For example, if on my page I have a login form, and I use the alert to inform him of an error in his login / password, VoiceOver sees the connection error, but he always sees the page with the login form.
This behavior is disturbing for users, hence my idea of adding an aria-modal = "true", so that the alert window is the only one visible on the screen by VoiceOver.
Logically, VoiceOver should even be able to speak the alert message automatically, without having to look for the alert window that is currently visible at the end of the page.
Thank you

**Comment by liamdebeasi** [comment]
Thanks for the follow up. Can you try this again with Ionic Framework 5.1? In that update we automatically focus the `ion-alert` element when presented, so VoiceOver should see that initially. In addition, it looks like we already have `aria-modal="true"` set on the `ion-alert`: https://github.com/ionic-team/ionic/blob/master/core/src/components/alert/alert.tsx#L468-L469.

**Comment by terrancle** [comment]
Hello,

Indeed, the focus is automatically positioned on an alert with Ionic 5.1, which is very good news.

However, it would be good to add this property, to be able to choose according to the situation, if the aria-modal should be activated or not.

For example, it could be false on ion-loading, so that the focus is not automatically positioned on a loading message.

The main thing would be that there is a choice.

Thank you !

**Comment by liamdebeasi** [comment]
Thanks for the clarification! Can you provide a use case for this? When would a screen reader not want to be informed that something is loading?

**Comment by terrancle** [comment]
I think we have to think about the priority of the posted content.
An alert must be in the foreground from the moment its information is given priority. For example "are you sure you want to delete this item?" is important.
In the case of an ion-loading, I do not think that the focus of the alert is essential.
It is for this reason that I propose to add a property in the definition of ion-alert, so that developers can have the choice of aria-modal to true or false.
For ion-loading, this property could be disabled by default, since the loading message does not have priority importance with each page loading (the message is available at the end of the page).

**Comment by liamdebeasi** [comment]
That makes sense thanks. In this case, `ion-loading` takes up the entire screen, and users should not be able to interact with any other part of the application while the loader is displayed. I think it might be confusing for those relying on screen readers to not know about the presence of `ion-loading` while also not being able to interact with the page. Thoughts?

**Comment by terrancle** [comment]
Hello,

You are right. I'll wait for the feedback from my users to see what they think.
The only thing might be to add the choice to the developers, for aria-modal true or false.
Depending on the situation, the developer will have a choice.
Above all, do not change the behavior of Ionic 5.1, because it is really good for 95% of situations.

**Comment by liamdebeasi** [comment]
Ok that sounds good. I am going to close the issue for now, but if you receive new feedback from users please feel free to open a new issue. Thanks!
